core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -288,9 +288,7 @@ static void addBindPermissions(Permissions policy, Settings settings) throws IOE
         for (Map.Entry<String, Settings> entry : profiles.entrySet()) {
             Settings profileSettings = entry.getValue();
             String name = entry.getKey();
-            String transportRange = profileSettings.get("port",
-                                        settings.get("transport.tcp.port",
-                                                NettyTransport.DEFAULT_PORT_RANGE));
+            String transportRange = profileSettings.get("port",NettyTransport.PORT.get(settings));
```

Comment:
> Also, for my understanding - what is the difference in terms of class loading between running that static `NettyTrasnport.DEFAULT_RANGE` that was before?

Accessing static compile-time constant variables will not trigger class initialization, but any other access to a class member will trigger class initialization immediately before the first such use (the JLS provides very strong guarantees about [when](https://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.1) and [how](https://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.2) this occurs).

> This is so easy to slip without anyone noticing.

Yes. The [suggestion](https://github.com/elastic/elasticsearch/pull/14549/files#r44182190) for a test for this has come up before.